src/store/data-store-level.ts

```diff
@@ -63,9 +63,11 @@ export class DataStoreLevel implements DataStore {
     const asyncDataBlocks = importer([{ content: dataStream }], blocks, { cidVersion: 1 });
 
     // NOTE: the last block contains the root CID as well as info to derive the data size
-    let dataDagRoot: ImportResult;
+    let dataDagRoot: ImportResult | undefined = undefined;
     for await (dataDagRoot of asyncDataBlocks) { ; }
 
+    if (dataDagRoot == undefined) { return { dataCid, dataSize: 0 }; } // FIXME: Is this a valid empty PutResult?
```

Comment:
@thehenrytsai Curious what your thoughts are here. IIUC, `dataDagRoot` should only be undefined if there is an issue getting `asyncDataBlocks` from IPFS. In that case, we should throw an exception. Does that sound right to you?

Reply:
Resolving since this can be addressed in a future PR if it becomes important.